src/com/palmergames/bukkit/towny/listeners/TownyEntityListener.java

```diff
@@ -375,6 +378,39 @@ public void onEntityDamageByEntityEvent(EntityDamageByEntityEvent event) {
 			
 	}
 
+	/**
+	 * Handles dragon fireball's cloud damage to players.
+	 * 
+	 * @param event AreaEffectCloudApplyEvent
+	 */
+	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+	public void onDragonFireBallCloudDamage(AreaEffectCloudApplyEvent event) {
+		if (!TownyAPI.getInstance().isTownyWorld(event.getEntity().getWorld()))
+			return;
+		
+		if (!event.getEntity().getBasePotionData().getType().equals(PotionType.UNCRAFTABLE))
+			return;
+
+		if (!(event.getEntity().getSource() instanceof Player) || !(event.getEntity().getSource() instanceof DragonFireball))
+			return;
+
+		List<LivingEntity> entities = event.getAffectedEntities();
+
+		TownyWorld townyWorld = null;
+		try {
+			townyWorld = TownyUniverse.getInstance().getDataSource().getWorld(event.getEntity().getWorld().getName());
+		} catch (NotRegisteredException e) {
+			// Failed to fetch a world
+			return;
+		}
+		
+		for (LivingEntity entity : entities) {
+			TownBlock townBlock = TownyAPI.getInstance().getTownBlock(entity.getLocation());
+			if (CombatUtil.preventPvP(townyWorld, townBlock)) {
+				event.setCancelled(true);

```

Comment:
I would cancel and do event.getEntity().remove() this way the same checks aren't being done every 5 ticks.

Reply:
Yeah, not all entities might be immune from the cloud though, some could be in a PVP-enabled area.